staging/src/k8s.io/client-go/tools/cache/thread_safe_store.go

```diff
@@ -330,3 +406,12 @@ func NewThreadSafeStore(indexers Indexers, indices Indices) ThreadSafeStore {
 		indices:  indices,
 	}
 }
+
+// StorageWithJointIndexer covariant ThreadSafeStore to ThreadSafeStoreWithJointIndexer
+func StorageWithJointIndexer(threadSafeStore ThreadSafeStore) (ThreadSafeStoreWithJointIndexer, error) {
```

Comment:
This function is not used anywhere. Also see my question on why don't we simply use golang's type assertion here?